src/main/java/io/leonis/subra/math/LimitPlayerCommand.java

```diff
@@ -0,0 +1,39 @@
+package io.leonis.subra.math;
+
+import io.leonis.subra.game.data.PlayerCommand;
+import lombok.experimental.Delegate;
+
+/**
+ * The Class LimitPlayerCommand.
+ *
+ * This class represents the computation of limiting the velocity in a {@link PlayerCommand}.
+ *
+ * @author Rimon Oz
+ */
+public class LimitPlayerCommand implements PlayerCommand {
+  @Delegate
+  private final PlayerCommand playerCommand;
+
+  /**
+   * Computes the {@link PlayerCommand} as a result of limiting the velocity vector by the supplied
+   * speed limit.
+   *
+   * @param command    The {@link PlayerCommand} to limit.
+   * @param speedLimit The speed limit.
+   */
+  public LimitPlayerCommand(final PlayerCommand command, final float speedLimit) {
```

Comment:
I think we might need to use past tense for the player commands ?

Reply:
what do you mean by past tense

Reply:
`LimitedPlayerCommand`, `AddedPlayerCommand`. But while i strongly prefer `limitedPlayerCommand`, i don't prefer `addedPlayerCommand` so we're probably fine using the current naming.

Reply:
that's not past tense, it's a past participle (adjective). i think present tense imperative works better because it indicates what instantiating the object will do. whereas using past participle indicates what instantiating the object did, and what it did cannot be deduced from the contents of the object